include/openthread/srp_client.h

```diff
@@ -0,0 +1,501 @@
+ *
+ * @retval OT_ERROR_NONE       SRP client operation started successfully or it is already running with same server
+ *                             socket address and callback.
+ * @retval OT_ERROR_BUSY       SRP client is busy running with a different socket address and/or callback.
+ * @retval OT_ERROR_FAILED     Failed to open/connect the client's UDP socket.
+ *
+ */
+otError otSrpClientStart(otInstance *        aInstance,
+                         const otSockAddr *  aServerSockAddr,
+                         otSrpClientCallback aCallback,
+                         void *              aContext);
+
+/**
+ * This function stops the SRP client operation.
+ *
+ * This function stops any further interactions with the SRP server. Note that it does not remove or clear host info
+ * and/or list of services. It marks all services to be added/removed again once the client is (re)started.
+ *
+ * @param[in] aInstance       A pointer to the OpenThread instance.
+ *
+ */
+void otSrpClientStop(otInstance *aInstance);
+
+/**
+ * This function gets the lease interval used in SRP update requests.
+ *
+ * Note that this is the lease duration requested by the SRP client. The server may choose to accept a different lease
+ * interval.
+ *
+ * @param[in] aInstance        A pointer to the OpenThread instance.
+ *
+ * @returns The lease interval (in seconds).
+ *
+ */
+uint32_t otSrpClientGetLeaseInterval(otInstance *aInstance);
+
+/**
+ * This function sets the lease interval used in SRP update requests.
+ *
+ * Changing the lease interval does not impact the accepted lease interval of already registered services/host-info.
+ * It only affects any future SRP update messages (i.e., adding new services and/or refreshes of the existing services).
+ *
+ * @param[in] aInstance   A pointer to the OpenThread instance.
+ * @param[in] aInterval   The lease interval (in seconds).
+ *
+ */
+void otSrpClientSetLeaseInterval(otInstance *aInstance, uint32_t aInterval);
+
+/**
+ * This function gets the key lease interval used in SRP update requests.
+ *
+ * Note that this is the lease duration requested by the SRP client. The server may choose to accept a different lease
+ * interval.
+ *
+ * @param[in] aInstance        A pointer to the OpenThread instance.
+ *
+ * @returns The key lease interval (in seconds).
+ *
+ */
+uint32_t otSrpClientGetKeyLeaseInterval(otInstance *aInstance);
+
+/**
+ * This function sets the key lease interval used in SRP update requests.
+ *
+ * Changing the lease interval does not impact the accepted lease interval of already registered services/host-info.
+ * It only affects any future SRP update messages (i.e., adding new services and/or refreshes of existing services).
+ *
+ * @param[in] aInstance    A pointer to the OpenThread instance.
+ * @param[in] aInterval    The key lease interval (in seconds).
+ *
+ */
+void otSrpClientSetKeyLeaseInterval(otInstance *aInstance, uint32_t aInterval);
```

Comment:
Is there any spec text that states this restriction?

Reply:
I don't recall if this is mentioned in the spec. It seems reasonable though.

That said, I think/suggest that it would be better to require server to check this.

 I intentionally skipped and/or removed many corner-case or extra checks  in the client implementation to reduce its code size (having in mind that this code may need to run on resource limited platforms).

Reply:
Yes, the server side should check the values. Okay to leave the check to reduce code size.